### ISSUE TITLE
Align jetty-schemas version within apache-jsp module as well

### ISSUE THREAD

**Issue by mohitj13** [eclipse/jetty.project#3104]
```
     - By conflict resolution : between versions 3.1.2 and 3.1

org.eclipse.jetty.toolchain:jetty-schemas:3.1.2
\--- org.eclipse.jetty:apache-jsp:9.4.13.v20181111
     \--- runtime

org.eclipse.jetty.toolchain:jetty-schemas:3.1 -> 3.1.2
\--- org.mortbay.jasper:apache-jsp:8.5.33
     \--- org.eclipse.jetty:apache-jsp:9.4.13.v20181111
          \--- runtime
```

Seems there is cyclic version conflict within apache-jsp

**Comment by olamy** [comment]
@mohitj13 can we have the output of `mvn dependency:tree` for your project.
The best would be a simple project to reproduce?

**Comment by gregw** [comment]
I'm not exactly sure why this is a problem (and not fixed by maven), but the apache-jsp jar still does depend on 3.1 schemas rather than 3.1.2.
I have updated with https://github.com/jetty-project/jasper-jsp/commit/acf08fde5b95db266a1c7ff8d71c6793a883d518, so if we are re-rolling 9.4.14, we should probably push a new version of the jsp artefacts.

**Comment by janbartel** [comment]
@mohitj13 there cannot be a cyclic dependency because  jetty-schemas does not depend on anything at all, and the org.mortbay.apache-jsp does not depend on anything in org.eclipse.jetty.  As @gregw has said, the org.mortbay.apache-jsp module at version 8.5.33 has a dependency on jetty-schemas 3.1, but the org.eclipse.jetty.apache-jsp module uses the newer 3.1.2 version, which maven resolves to use the higher 3.1.2 version.  

Please provide more details urgently as we want to begin the process of release for 9.4.14 ASAP.

**Comment by joakime** [comment]
This is not a maven issue.
The output format provided by OP is not maven.

**Comment by mohitj13** [comment]
Hi, My apologies for not describing this issue correctly on the first instance.

I have a Gradle Project, so "org.eclipse.jetty:apache-jsp:9.4.13.v20181111" is a dependency that I am loading into my project.

Now within project configuration I have explicitly stated that to fail on version conflict so that I am aware on the conflicted dependency.

I have attached a sample project which all it does is load this dependency. Try importing this project you will get the error.
[test-jsp.zip](https://github.com/eclipse/jetty.project/files/2580770/test-jsp.zip)
  

The output in the previous comment can be achieved by using command

> gradle :dependencyInsight --configuration runtime --dependency org.eclipse.jetty.toolchain:jetty-schemas

**Comment by janbartel** [comment]
@mohitj13 thanks for the info.  I believe we've solved the problem with PR #3108 which has now been applied to the jetty-9.4.x branch. It would be great if you could checkout that branch, build locally and then rerun your gradle build to verify the fix.

**Comment by janbartel** [comment]
We believe we fixed this with PR #3108, in the absence of any contradictory info from the OP, closing this issue.
